app/src/main/java/tech/relaycorp/letro/repository/GatewayRepository.kt

```diff
@@ -62,6 +84,163 @@ class GatewayRepository @Inject constructor(
+                recipientEndpoint = thirdPartyEndpoint,
+            )
+
+            GatewayClient.sendMessage(message)
+        }
+    }
+
+    fun startPairingWithContact(pairingRequestVeraIds: PairingRequestVeraIdsDataModel) {
+        gatewayScope.launch {
+            val firstPartyEndpoint = loadNonNullFirstPartyEndpoint(serverFirstPartyEndpointNodeId.value)
+            val thirdPartyEndpoint = loadNonNullThirdPartyEndpoint(serverThirdPartyEndpointNodeId.value)
+
+            val pairingRequestContent: ByteArray = generatePairingRequest(
+                pairingRequestVeraIds.requesterVeraId,
+                pairingRequestVeraIds.contactVeraId,
+                firstPartyEndpoint,
+            )
+
+            val pairingRequestMessage = OutgoingMessage.build(
+                ContentType.ContactPairingRequest.value,
+                pairingRequestContent,
+                firstPartyEndpoint,
+                thirdPartyEndpoint,
+            )
+
+            GatewayClient.sendMessage(pairingRequestMessage)
+            _pairingRequestSent.emit(pairingRequestVeraIds)
+        }
+    }
+
+    fun sendPairingAuthorizationRequest(match: PairingMatchDataModel) {
+        gatewayScope.launch {
+            val firstPartyEndpoint = loadNonNullFirstPartyEndpoint(serverFirstPartyEndpointNodeId.value)
+            val thirdPartyEndpoint = loadNonNullThirdPartyEndpoint(serverThirdPartyEndpointNodeId.value)
+
+            val pairingAuth = generatePairingAuth(match, firstPartyEndpoint)
+
+            val pairingAuthMessage = OutgoingMessage.build(
+                ContentType.ContactPairingAuthorization.value,
+                pairingAuth,
+                firstPartyEndpoint,
+                thirdPartyEndpoint,
+            )
+
+            GatewayClient.sendMessage(pairingAuthMessage)
+            _pairingAuthorizationSent.emit(match)
+        }
+    }
+
+    private fun startReceivingMessages() {
+        gatewayScope.launch {
+            GatewayClient.receiveMessages().collect { message ->
+                // TODO Remove first message.ack() before publishing the app.
+                // It's here to avoid the server getting stuck with messages that can't be processed.
+                message.ack()
+                when (message.type) {
+                    ContentType.AccountCreationCompleted.value -> {
+                        val veraIds =
+                            message.content.toString(Charset.defaultCharset()).split(",")
+                        _accountCreatedConfirmationReceivedFromServer.emit(
+                            AccountCreatedDataModel(
+                                veraIds[0],
+                                veraIds[1],
+                            ),
+                        )
+                        message.ack()
+                    }
+                    ContentType.ContactPairingMatch.value -> {
+                        // TODO Simplify this by moving to the ContactRepository
```

Comment:
@gnarea I thought about this while running through changes again. I would still keep this slightly more complicated writing for 2 reasons:
- follows the same principle for every message
- GatewayRepository is the only one "working" with Awala SDK

If you still think it should be simplified, then I would do the change for every message received. Please, let me know your thoughts